src/MusicTheory/Duration.elm

```diff
@@ -0,0 +1,37 @@
+module MusicTheory.Duration exposing
+    ( Division(..)
+    , Duration(..)
+    , Grouping(..)
+    , TiedOrSeparate(..)
+    )
+
+
+type Grouping
+    = Single Duration
+    | Duplet Duration Duration
+    | Triplet Duration Duration Duration
+    | Quadruplet Duration Duration Duration Duration
+    | Quintuplet Duration Duration Duration Duration Duration
```

Comment:
Probably obvious, but there has to be a way to put a note into the tuplets.

Or would you have the rhythm as a separate thing that can be zipped with a series of pitches, rests, etc.?

Reply:
Great question. I am not sure. What do you think the pros and cons are to each one?

Reply:
I went with combining them into records. This way the duration of a note and its correspondinng pitch(es) are always in sync with each other.

Reply:
I've seen the zipping approach. I don't like it. So this is fine.